core/src/org/pentaho/di/core/Const.java

```diff
@@ -2420,7 +2444,13 @@ public static int indexOfString( String lookup, List<String> list ) {
             addSplit = oddNumberOfEnclosures;
           }
           if ( addSplit ) {
-            splitList.add( concatSplit.toString() );
+            String splitResult = concatSplit.toString();
+            //remove enclosure from resulting split
+            if ( removeEnclosure ) {
+              splitResult = splitResult.replaceAll( enclosure, "" );
```

Comment:
@pamval As specified in original method's javadoc: "This method expects that the data contains an even number of enclosure strings in the input; otherwise the results are undefined". 
So before fix it worked like:
"abc", "de" -> ["abc"] ["de"]
"ab"c", "de" -> [] []
"abc", "d"e" -> [] ["de"]

After fix: 
"abc", "de" -> [abc] [de]
"ab"c", "de" -> [] []
"abc", "d"e" -> [] [de]